File: lib/domain/models/score_result.dart
class ScoreResult {
  final int basePoints;
  final bool isParent;
  final bool isTsumo;

  const ScoreResult({
    required this.basePoints,
    required this.isParent,
    required this.isTsumo,
  });

  int get ronPoints {
    final multiplier = isParent ? 6 : 4;
    return _roundUp(basePoints * multiplier, 100);
  }

  int get tsumoKoPoints => _roundUp(basePoints, 100);

  int get tsumoOyaPoints => _roundUp(basePoints * 2, 100);

  String toAnswerString() {
    if (isTsumo) {
      if (isParent) {
        return '$tsumoOyaPoints all';
      } else {
        return '$tsumoKoPoints/$tsumoOyaPoints';
      }
    } else {
      return '$ronPoints';
    }
  }

  static int _roundUp(int value, int unit) {
    return ((value + unit - 1) ~/ unit) * unit;
  }
}

File: src/domain/models/scoreResult.ts
export class ScoreResult {
  readonly basePoints: number;
  readonly isParent: boolean;
  readonly isTsumo: boolean;

  constructor(basePoints: number, isParent: boolean, isTsumo: boolean) {
    this.basePoints = basePoints;
    this.isParent = isParent;
    this.isTsumo = isTsumo;
  }

  get ronPoints(): number {
    const multiplier = this.isParent ? 6 : 4;
    return ScoreResult.roundUp(this.basePoints * multiplier, 100);
  }

  get tsumoKoPoints(): number {
    return ScoreResult.roundUp(this.basePoints, 100);
  }

  get tsumoOyaPoints(): number {
    return ScoreResult.roundUp(this.basePoints * 2, 100);
  }

  toAnswerString(): string {
    if (this.isTsumo) {
      if (this.isParent) {
        return `${this.tsumoOyaPoints} all`;
      } else {
        return `${this.tsumoKoPoints}/${this.tsumoOyaPoints}`;
      }
    } else {
      return `${this.ronPoints}`;
    }
  }

  static roundUp(value: number, unit: number): number {
    return Math.ceil(value / unit) * unit;
  }
}
